### PR TITLE
Turns the "set key for client control" pattern into a macro

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request 

This is the pattern:

```dm
subject.key = target.key
```

Where `subject` is the mob that we are currently working on, and `target` is an already-active mob with a client who we want to take control of `subject`. This pattern always makes me freaked out because I don't like working with keys, but it's a pattern that we must deal with. So, I decided to turn it into a macro that just expands to the same thing `KEY_TRANSFER()`. This has a few good upsides.

* Easier to grep
* If it breaks, easier to just expand the macro rather than change all the instances in the codebase
* If a better way to "transfer control" is figured out, we can just sub that in (refer to point `#1`)
* Able to document just exactly why we're transferring the keys in the first place (since this pattern might not be entirely familiar)
* If we ever need to do more stuff with key transfers (like standardized logging), we can easily just plonk it into this macro and move on with our lives
* Looks a bit cleaner overall, and solidifies that this is a valid pattern (at least until it eventually breaks (see point `#2`)).

Basically we just do this now:

```dm
TRANSFER_KEY(subject, target) // easy to ctrl+click the define to access all that documentation goodness
```
## Why It's Good For The Game

Code improvement.
## Changelog
Nothing that concerns players.

Let me know if you want a change to the name of the macro or something, shouldn't be too hard.

### CODE REVIEW

**Review by Mothblocks — CHANGES_REQUESTED** [review]
image

Closing unless anyone disagrees
